src/Mesh/Interpolation.jl

```diff
@@ -0,0 +1,472 @@
+module Interpolation
+
+using MPI
+import GaussQuadrature
+using CLIMA.Mesh.Topologies
+using CLIMA.Mesh.Grids
+using CLIMA.Mesh.Geometry
+using CLIMA.Mesh.Elements
+using LinearAlgebra
+using StaticArrays
+
+export InterpolationBrick, interpolate_brick!, interpolate_brick_v2!, InterpolationCubedSphere, invert_trilear_mapping_hex, interpolate_cubed_sphere!
+
+#--------------------------------------------------------
+"""
+    function InterpolationBrick(grid::DiscontinuousSpectralElementGrid, xres, ::Type{FT}) where FT <: AbstractFloat
+
+This interpolation structure and the corresponding functions works for a brick, where stretching/compression happens only along the x, y & z axis.
+Here x1 = X1(ξ1); x2 = X2(ξ2); x3 = X3(ξ3)
+
+# input for the inner constructor
+ - `grid` DiscontinousSpectralElementGrid
+ - `xres` Resolution of the interpolation grid in x1, x2 and x3 directions
+"""
+struct InterpolationBrick{FT <:AbstractFloat}
+    realelems::UnitRange{Int64}
+
+    xbnd::Array{FT}        # domain bounds, [2(min/max),ndim]
+    xres::Array{FT}        # resolutions in x1, x2, x3 directions for the uniform grid
+
+    ξ1::Vector{Vector{FT}} # unique ξ1 coordinates of interpolation points within each element 
+    ξ2::Vector{Vector{FT}} # unique ξ2 coordinates of interpolation points within each element 
+    ξ3::Vector{Vector{FT}} # unique ξ3 coordinates of interpolation points within each element 
+  
+    x::Vector{Array{FT}}   # x[elno][3,npts] -> (x1,x2,x3) coordinates of portion of interpolation grid embedded within each element 
+
+    V::Vector{Vector{FT}}  # interpolated variable within each element
+#--------------------------------------------------------
+    function InterpolationBrick(grid::DiscontinuousSpectralElementGrid, xres, ::Type{FT}) where FT <: AbstractFloat
```

Comment:
```suggestion
    function InterpolationBrick(grid::DiscontinuousSpectralElementGrid{FT}, xres) where FT <: AbstractFloat
```
You'll need to change the way this is called, too.